document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -24,14 +26,42 @@ public DocumentInfo getDocumentInfo() {
         LOG.info("Started getting document");
 
         String resource = "";
+        String resourceId = "";
 
         Transaction transaction = transactionService.getTransaction(resource);
         if (transaction == null) {
             LOG.error("transaction not found");
             return null;
         }
 
-        return new DocumentInfo();
+        String accountLink = Optional.of(transaction)
+                .map(Transaction::getResources)
+                .map(e -> e.get(resourceId))
+                .map(Resource::getLinks)
+                .map(e -> e.get("resource"))
+                .orElseGet(() -> {
+                    LOG.error("Unable to find account resource link in transaction under:" + resourceId);
+                    return Optional.empty().toString();
+                });
 
+        // when abridged has been migrated to use the company-accounts api, the code for the
+        // company accounts should work for abridged, resulting in this abridged specific code
+        // qualifying for removal
+        if (isAbridged(accountLink)) {
```

Comment:
Only a small suggestion - could add this to the optional which itself could be returned.
```
return Optional.of(transaction)
              ...
              ...
              .filter(this::isAbridged)
              .map(accountLinks -> new DocumentInfo())
              .orElse(null);
```

Reply:
Yeah good shout :+1: